### PR TITLE
IGNITE-18189: Merge ignite-spark 3.2 and 2.4 extensions into a single…

### PR DESCRIPTION
**Pull request by ashapkin**

### CODE REVIEW

**Comment by Mmuzaf** [comment]
`/core/src/test/config/log4j2-test.xml`
`core/src/test/config/tests.properties`

Please, remove these files. They are not required anymore.

File: modules/spark-ext/examples/pom.xml [hunk 1]
```diff
@@ -29,42 +29,72 @@
...
 
     <dependencies>
+
         <dependency>
-            <groupId>org.apache.ignite</groupId>
-            <artifactId>ignite-core</artifactId>
-            <version>${ignite.version}</version>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>ignite-spark-ext</artifactId>
+            <version>${project.version}</version>
```
Comment (Mmuzaf) [inline modules/spark-ext/examples/pom.xml:36]: This must be `${project.parent.version}` instead.

**Review by Mmuzaf — CHANGES_REQUESTED** [review]

File: modules/spark-ext/spark/src/main/scala/org/apache/spark/sql/ignite/IgniteSparkSession.scala [hunk 2]
```diff
@@ -142,7 +142,7 @@ class IgniteSparkSession private(
     override def createDataset[T: Encoder](data: Seq[T]): Dataset[T] = {
         val enc = encoderFor[T]
         val attributes = enc.schema.toAttributes
-        val encoded = data.map(d => enc.toRow(d).copy())
+        val encoded = data.map(d => enc.createSerializer().apply(d)) // TODO .copy()?
```
[the remark this answers is not in the document]
Comment (ashapkin) [reply modules/spark-ext/spark/src/main/scala/org/apache/spark/sql/ignite/IgniteSparkSession.scala:145]: Unfortunately, I'm not the author of that change, so I'd better remove them.

File: modules/spark-ext/spark/src/main/scala/org/apache/spark/sql/ignite/IgniteExternalCatalog.scala [hunk 3]
```diff
@@ -318,6 +318,12 @@ private[ignite] class IgniteExternalCatalog(igniteContext: IgniteContext)
     override def alterPartitions(db: String, table: String,
         parts: Seq[CatalogTablePartition]): Unit =
         throw new UnsupportedOperationException("unsupported")
+
+    override def getTablesByName(db: String, tables: Seq[String]): Seq[CatalogTable] = tables.map(getTable(db, _))
+
+    override def listViews(db: String, pattern: String): Seq[String] = {
+        throw new UnsupportedOperationException("unsupported") // FIXME
```
[the remark this answers is not in the document]
Comment (ashapkin) [reply modules/spark-ext/spark/src/main/scala/org/apache/spark/sql/ignite/IgniteExternalCatalog.scala]: I'm not sure, let me clean that type of comment and ask the developers to keep an eye on it.

File: modules/spark-ext/pom.xml [hunk 4]
```diff
@@ -34,12 +34,10 @@
     <version>2.0.0-SNAPSHOT</version>
```
[the remark this answers is not in the document]
Comment (ashapkin) [reply modules/spark-ext/pom.xml]: Done

**Review by Mmuzaf — APPROVED** [review]

**Pull request merged**